protobuf/endpoints/register_function.proto

```diff
@@ -4,8 +4,55 @@ package endpoints;
 
 import "models/registered_function.proto";
 
+message CpuRequirements {
+  // Optional field where 1000 millicores equals 1 CPU.
+  // From https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu:
+  // "For CPU resource units, the quantity expression 0.1 is equivalent to the expression 100m, which can be read as 'one hundred millicpu'."
+  optional uint32 millicores = 1;
+
+  // Optional field indicating amount of memory needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 memory_in_bytes = 2;
+
+  // Optional field indicating amount of storage needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 storage_in_bytes = 3;
+}
+
+message GpuRequirements {
```

Comment:
Do we support the jobs requiring GPUs with some computation offloaded to the CPU?